protocol/chainlib/chain_router.go

```diff
@@ -114,10 +117,51 @@ func (cri *chainRouterImpl) BatchNodeUrlsByServices(rpcProviderEndpoint lavasess
 				methodRoutes[methodRoutesUnique] = len(methodRoutes)
 				existing = len(methodRoutes)
 			}
-			routerKey = routerKey.ApplyMethodsRoute(existing)
+			routerKey.ApplyMethodsRoute(existing)
+		}
+		routerKey.ApplyInternalPath(nodeUrl.InternalPath)
+		isWs, err := IsUrlWebSocket(nodeUrl.Url)
+		// Some parsing may fail because of gRPC
+		if err == nil && isWs {
+			// save the first ws router key and nodeUrl for later use
+			firstWsRouterKey = routerKey
```

Comment:
you are overwriting this every time, not sure if this is intended, if so maybe call this lastWsRouterKey?

Reply:
Fixed in a8f7367d813366f78fbd459bd1d54ee3fe2eb53e